scipy/stats/tests/test_stats.py

```diff
@@ -2875,41 +2876,112 @@ def test_friedmanchisquare():
 
-    # test for namedtuple attribute results
-    attributes = ('statistic', 'pvalue')
-    res = stats.kstest(x, 'norm')
-    check_named_results(res, attributes)
+    def _test_kstest_and_ks1samp(self, x, alternative, mode='auto', decimal=14):
+        result = stats.kstest(x, 'norm', alternative=alternative, mode=mode)
+        result_1samp = stats.ks_1samp(x, stats.norm.cdf, alternative=alternative, mode=mode)
+        assert_array_almost_equal(np.array(result), result_1samp, decimal=decimal)
+
+    def test_namedtuple_attributes(self):
+        x = np.linspace(-1, 1, 9)
+        # test for namedtuple attribute results
+        attributes = ('statistic', 'pvalue')
+        res = stats.kstest(x, 'norm')
+        check_named_results(res, attributes)
+
+    def test_agree_with_ks_1samp(self):
+        x = np.linspace(-1,1,9)
+        self._test_kstest_and_ks1samp(x, 'two-sided')
+
+        x = np.linspace(-15,15,9)
+        self._test_kstest_and_ks1samp(x, 'two-sided')
+
+        x = [-1.23, 0.06, -0.60, 0.17, 0.66, -0.17, -0.08, 0.27, -0.98, -0.99]
+        self._test_kstest_and_ks1samp(x, 'two-sided')
+        self._test_kstest_and_ks1samp(x, 'greater', mode='exact')
+        self._test_kstest_and_ks1samp(x, 'less', mode='exact')
+
+    # missing: no test that uses *args
 
-    # the following tests rely on deterministicaly replicated rvs
-    np.random.seed(987654321)
-    x = stats.norm.rvs(loc=0.2, size=100)
-    D,p = stats.kstest(x, 'norm', mode='asymp')
-    assert_almost_equal(D, 0.12464329735846891, 15)
-    assert_almost_equal(p, 0.089444888711820769, 15)
-    assert_almost_equal(np.array(stats.kstest(x, 'norm', mode='asymp')),
-                np.array((0.12464329735846891, 0.089444888711820769)), 15)
-    assert_almost_equal(np.array(stats.kstest(x,'norm', alternative='less')),
-                np.array((0.12464329735846891, 0.040989164077641749)), 15)
-    # this 'greater' test fails with precision of decimal=14
-    assert_almost_equal(np.array(stats.kstest(x,'norm', alternative='greater')),
-                np.array((0.0072115233216310994, 0.98531158590396228)), 12)
+class TestKSOneSample(object):
+    """Tests kstest and ks_samp 1-samples with K-S various sizes, alternatives, modes."""
+
+    def _testOne(self, x, alternative, expected_statistic, expected_prob, mode='auto', decimal=14):
+        result = stats.ks_1samp(x, stats.norm.cdf, alternative=alternative, mode=mode)
+        expected = np.array([expected_statistic, expected_prob])
+        assert_array_almost_equal(np.array(result), expected, decimal=decimal)
+
+    def test_namedtuple_attributes(self):
+        x = np.linspace(-1, 1, 9)
+        # test for namedtuple attribute results
+        attributes = ('statistic', 'pvalue')
+        res = stats.ks_1samp(x, stats.norm.cdf)
+        check_named_results(res, attributes)
+
+    def test_agree_with_r(self):
+        # comparing with some values from R
+        x = np.linspace(-1,1,9)
```

Comment:
do the missing spaces after , cause the whitespace error? there are a few more cases above and below